website/src/pages/routes/routesPage.js

```diff
@@ -55,25 +51,32 @@ function RoutesPage(props) {
     list.addEventListener("click", async (event) => {
         const id = event.target.dataset.id
         const route = await routesService.findById(id)
+        //console.log(route.pointsOfInterest[0].name)
 
-        console.log(route.pointsOfInterest[0].name)
-
-        const coordinates = route.pointsOfInterest.map((point)=> {
+        const coordinates = await route.pointsOfInterest.map((point)=> {
             const lat = point.location.latitude
             const long = point.location.longitude
-            return [{lat}, {long}]
+            return [long, lat]
         })
-        // so far so good? is an array of two doubles objects the right format for coordinates???
         console.log(coordinates)
-        await loadRoute(coordinates)
-    });
+        const coord2 = [[15.188356982912637, 55.320417209601885], [15.1928236, 55.3201917]]
+
+        if (map.getLayer('route')) {
+            map.removeLayer('route')
+            map.removeSource('route')
+        }
+
+        loadRoute(coordinates)
+
+    })
 
-    async function loadRoute (...coords) {
-        map.resize();
-        const waypoints = await mapService.getRouteFromCoordinates(
-            [15.188356982912637, 55.320417209601885], [15.1928236, 55.3201917]
+    async function loadRoute (coordinates) {
+        //map.resize();
+        const waypoints = await getRouteFromCoordinates(
+            coordinates
```

Comment:
Using the [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), each element in the array can be spread out as individual parameters for the function.
```suggestion
            ...coordinates
```

Reply:
Your suggestion breaks the code. As we already talked about, it is just the fix we found with Asger to make the code work!